service-app/internal/api/handler.go

```diff
@@ -20,31 +21,75 @@ import (
 )
 
 type IndexController struct {
-	config    *config.Config
-	logger    *logger.Logger
-	validator *ingestion.Validator
-	Queue     *ingestion.JobQueue
-	AwsClient *aws.AwsClient
+	config         *config.Config
+	logger         *logger.Logger
+	validator      *ingestion.Validator
+	httpMiddleware *httpclient.Middleware
+	authMiddleware *auth.Middleware
+	Queue          *ingestion.JobQueue
+	AwsClient      *aws.AwsClient
 }
 
 func NewIndexController(awsClient *aws.AwsClient, appConfig *config.Config) *IndexController {
+	logger := logger.NewLogger(appConfig)
+
+	// Create dependencies
+	httpClient := httpclient.NewHttpClient(*appConfig, *logger)
+	tokenGenerator := auth.NewJWTTokenGenerator(awsClient, appConfig, logger)
+	cookieHelper := auth.MembraneCookieHelper{
+		CookieName: "membrane",
+		Secure:     appConfig.App.Environment != "local",
+	}
+	authenticator := auth.NewBasicAuthAuthenticator(awsClient, cookieHelper, tokenGenerator)
+
+	// Create authentication middleware
+	authMiddleware := auth.NewMiddleware(authenticator, tokenGenerator, cookieHelper, logger)
+	// Create HTTP middleware
+	httpMiddleware, _ := httpclient.NewMiddleware(httpClient, tokenGenerator)
+
 	return &IndexController{
-		config:    appConfig,
-		logger:    logger.NewLogger(appConfig),
-		validator: ingestion.NewValidator(),
-		Queue:     ingestion.NewJobQueue(appConfig),
-		AwsClient: awsClient,
+		config:         appConfig,
+		logger:         logger,
+		validator:      ingestion.NewValidator(),
+		httpMiddleware: httpMiddleware,
+		authMiddleware: authMiddleware,
+		Queue:          ingestion.NewJobQueue(appConfig),
+		AwsClient:      awsClient,
 	}
 }
 
 func (c *IndexController) HandleRequests() {
-	http.Handle("/ingest", telemetry.Middleware(c.logger.SlogLogger)(http.HandlerFunc(c.IngestHandler)))
+	// Create the /auth route to handle user authentication and issue JWT token
+	http.Handle("/auth", http.HandlerFunc(c.AuthHandler))
```

Comment:
Suggest that we align the naming here with the existing DDC api naming i.e. `/auth/sessions` for auth and` /api/ddc` for sets, just so there's minimum change needed on the supplier side.